File: 240116/그 요일은/the-day-of-the-day.java
import java.util.Scanner;
public class Main {
    public static int mon(int x, int y){
        int[] day = new int[]{0,31,29,31,30,31,30,31,31,30,31,30,31};
        int tal = 0 ;
        for (int i = 1; i<x; i++){
            tal +=day[i];
        }
        tal = tal+y;
        return tal;
    }
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt();
        int d1 = sc.nextInt();
        int m2 = sc.nextInt();
        int d2 = sc.nextInt();
        String word = sc.next();
        int diff = mon(m2,d2) - mon(m1,d1)+1;
        int cnt = 0 ;
        String[] day2 = new String[]{"Mon","Tue","Wed","Thu","Fri","Sat","Sun"};
        for (int i = 0 ; i<diff; i++){
            if (day2[i%7].equals(word)){
                cnt++;
            }
        } 
        System.out.print(cnt);
    }
}

File: 240116/그 요일은/the-day-of-the-day.py
def calculate_days(month, day):
    total = 0
    for i in range(1, month):
        total += month_days[i]
    total += day
    return total

m1, d1, m2, d2 = map(int, input().split())
word = input()

month_days = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
days_of_week = ["Mon", 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']

def get_day_index(day_name):
    return days_of_week.index(day_name)

start_day = get_day_index(word)
diff = abs(calculate_days(m1, d1) - calculate_days(m2, d2))

count = 0
for i in range(diff):
    if days_of_week[(start_day + i) % 7] == word:
        count += 1

print(count)